### PR TITLE
feat: simplify Lua generation by using Kong Gateway image

### PR DESCRIPTION
**Pull request by mikefero**
This feature simplifies the Lua generation for goks by using the released image of Kong Gateway OSS. This ensures that luarocks installation does not update any libraries that can be updated based on version matching for dependencies. This also removes the need to compile luarock dependencies (FFI) and reduces the update process time to seconds.

**Note**: Lua files were re-generated to account for missing file in lua-tree. This file it not currently in use by the metaschema validation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
-	docker build -t goks -f build/Dockerfile .
-	docker run -it -v $(PWD):/goks goks
+	@cd docker && \
+		docker build \
+			--build-arg GOKS_KONG_GATEWAY_VERSION=${GOKS_KONG_GATEWAY_VERSION} \
+			-t kong-gateway-oss-source \
+			-f Dockerfile.kong-gateway-oss-source \
+			.
+	@docker run \
+		-it \
```
Comment (javierguerragiraldez) [inline Makefile:14]: aren't these two flags ('i' and 't') useful only for interactivity?  they feel a bit weird on a script.

File: lua-tree/share/lua/5.1/kong/tools/batch_queue.lua [hunk 2]
```diff
@@ -0,0 +1,327 @@
+-- Batch queue with background retryable processing
```
Comment (javierguerragiraldez) [inline lua-tree/share/lua/5.1/kong/tools/batch_queue.lua:1]: is this file part of a different patch?  i don't see where it's called from.
(not an objection, just trying to understand where does it fit)

File: Makefile (hunk 1 above)
Comment (mikefero) [reply Makefile:14]: Good call, was cleaning up and left that in.

Comment (mikefero) [reply Makefile]: Squashed within original commit

**Comment by mikefero** [comment]
#justfyi: The last commit was from a different merge into this branch (already approved).

**Review by javierguerragiraldez — APPROVED** [review]

**Pull request merged**